File: src/main/java/com/citasmedicas/appcitasmedicas/Service/AppointmentService.java
package com.citasmedicas.appcitasmedicas.Service;

import com.citasmedicas.appcitasmedicas.dto.Request.CancelAppointmentRequest;
import com.citasmedicas.appcitasmedicas.dto.Request.CreateAppointmentRequest;
import com.citasmedicas.appcitasmedicas.dto.Response.AppointmentResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;


public interface AppointmentService {
    AppointmentResponse create(CreateAppointmentRequest request);
    AppointmentResponse findById(Long id);
    Page<AppointmentResponse> findAll(Pageable Page);
    AppointmentResponse confirm(Long id);
    AppointmentResponse cancel(Long id, CancelAppointmentRequest request);
    AppointmentResponse complete(Long id, String observations);
    AppointmentResponse markNoShow(Long id);
    void delete(Long id);
}

File: frontend/src/services/AppointmentService.js
import api from './api';

export const appointmentService = {
  getAll: (page = 0, size = 10) =>
    api.get(`/appointments?page=${page}&size=${size}`),

  getById: (id) =>
    api.get(`/appointments/${id}`),

  create: (data) =>
    api.post('/appointments', data),

  confirm: (id) =>
    api.put(`/appointments/${id}/confirm`),

  cancel: (id, cancellationReason) =>
    api.put(`/appointments/${id}/cancel`, { cancellationReason }),

  complete: (id, observations = null) =>
    api.put(`/appointments/${id}/complete`, observations ? { observations } : {}),

  markNoShow: (id) =>
    api.put(`/appointments/${id}/no-show`),

  update: (id, data) => 
    api.patch(`/appointments/${id}/`, data),

  delete: (id) =>
    api.delete(`/appointments/${id}`),
};